src/main/java/net/dries007/tfc/compat/wailatop/WailaBlocks.java

```diff
@@ -0,0 +1,502 @@
+            currenttip = PlacedItemFlatBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEPitKiln) currenttip = PitKilnBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TECharcoalForge)
+            currenttip = CharcoalForgeBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TELogPile) currenttip = LogPileBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEBarrel) currenttip = BarrelBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEBloomery) currenttip = BloomeryBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEBlastFurnace)
+            currenttip = BlastFurnaceBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEFirePit) currenttip = FirepitBody(itemStack, currenttip, accessor, config);
+            // Crops and Trees
+        else if (te instanceof TECropBase) currenttip = CropSimpleBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TECropSpreading)
+            currenttip = CropSpreadingBody(itemStack, currenttip, accessor, config);
+        else if (b instanceof BlockFruitTreeLeaves)
+            currenttip = FruitTreeLeavesBody(itemStack, currenttip, accessor, config);
+        else if (b instanceof BlockBerryBush) currenttip = BerryBushBody(itemStack, currenttip, accessor, config);
+        else if (b instanceof BlockFarmlandTFC)
+            currenttip = FarmlandTFCBody(itemStack, currenttip, accessor, config);
+        else if (b instanceof BlockCropDead) currenttip = CropDeadBody(itemStack, currenttip, accessor, config);
+        else if (b instanceof BlockFruitTreeLeaves)
+            currenttip = FruitTreeLeavesBody(itemStack, currenttip, accessor, config);
+
+        return currenttip;
+    }
+
+    @Override
+    public List<String> getWailaTail(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        if (config.getConfig("tfc.displayTemp"))
+        {
+            currenttip.add(new TextComponentTranslation("waila.temperature").getFormattedText() + " : " + String.valueOf(Math.round(ClimateTFC.getActualTemp(accessor.getWorld(), accessor.getPosition(), 0))) + new TextComponentTranslation("waila.tempsymbol").getFormattedText());
+        }
+        return currenttip;
+    }
+
+    public List<String> FirepitBody(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        return currenttip;
+    }
+
+    public List<String> PitKilnBody(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        Block b = accessor.getBlock();
+        TEPitKiln te = (TEPitKiln) accessor.getTileEntity();
+        String key;
+        Boolean isLit = te.isLit();
+        Long litTick = te.getLitTick();
```

Comment:
```suggestion
        long litTick = te.getLitTick();
```
Don't use boxed primitives unless nessecary.